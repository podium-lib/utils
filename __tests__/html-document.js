'use strict';

const HttpIncoming = require('../lib/http-incoming');
const document = require('../lib/html-document');

const SIMPLE_REQ = {
    headers: {},
};

const SIMPLE_RES = {
    locals: {},
};

/**
 * .document()
 */

test('.document() - no arguments given - should render template', () => {
    const incoming = new HttpIncoming(SIMPLE_REQ, SIMPLE_RES);
    const result = document(incoming);
    expect(result).toMatchSnapshot();
});

test('.document() - arguments given - should render template using values given', () => {
    const incoming = new HttpIncoming(SIMPLE_REQ, SIMPLE_RES);
    incoming.css = [{ value: 'http://somecssurl.com' }];
    incoming.js = [{ value: 'http://somejsurl.com' }];
    incoming.context = {
        locale: 'en-NZ',
    };
    incoming.view = {
        encoding: 'utf-pretend-encoding',
        title: 'this goes in the title tag',
    };

    const head = 'this goes in the head section';
    const body = 'this goes in the body section';

    const result = document(incoming, body, head);
    expect(result).toMatchSnapshot();
});

test('.document() - arguments given - handles v4 js and css syntax', () => {
    const incoming = new HttpIncoming(SIMPLE_REQ, SIMPLE_RES);
    incoming.css = [
        { value: 'http://somecssurl1.com', type: 'text/css' },
        { value: 'http://somecssurl2.com', type: 'text/css' },
        { value: 'http://somecssurl3.com', type: 'text/css' },
    ];
    incoming.js = [
        { value: 'http://somejsurl1.com', type: 'default' },
        { value: 'http://somejsurl2.com', type: 'default' },
        { value: 'http://somejsurl3.com', type: 'default' },
    ];

    const result = document(incoming);
    expect(result).toMatchSnapshot();
});

test('.document() - js "type" is "esm" - should set type to module on script tags', () => {
    const incoming = new HttpIncoming(SIMPLE_REQ, SIMPLE_RES);
    incoming.css = [
        { value: 'http://somecssurl1.com', type: 'text/css' },
        { value: 'http://somecssurl2.com', type: 'text/css' },
        { value: 'http://somecssurl3.com', type: 'text/css' },
    ];
    incoming.js = [
        { value: 'http://somejsurl1.com', type: 'esm' },
        { value: 'http://somejsurl2.com', type: 'esm' },
        { value: 'http://somejsurl3.com', type: 'esm' },
    ];

    const result = document(incoming);
    expect(result).toMatchSnapshot();
});

test('.document() - js "type" is "esm" - should set type to module on script tags', () => {
    const incoming = new HttpIncoming(SIMPLE_REQ, SIMPLE_RES);
    incoming.css = [{ value: 'http://somecssurl1.com', type: 'text/css' }];
    incoming.js = [
        { value: 'http://somejsurl1.com/lazy', type: 'module', strategy: 'lazy' },
        { value: 'http://somejsurl2.com/before', type: 'module', strategy: 'beforeInteractive' },
        { value: 'http://somejsurl3.com/after', type: 'module', strategy: 'afterInteractive' },
    ];

    const result = document(incoming);
    expect(result).toMatchSnapshot();
});
