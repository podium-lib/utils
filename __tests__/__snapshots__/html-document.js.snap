// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.document() - arguments given - handles v4 js and css syntax 1`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link href=\\"http://somecssurl1.com\\" type=\\"text/css\\">
        <link href=\\"http://somecssurl2.com\\" type=\\"text/css\\">
        <link href=\\"http://somecssurl3.com\\" type=\\"text/css\\">
        
        <title></title>
        
    </head>
    <body>
        
        <script src=\\"http://somejsurl1.com\\"></script>
        <script src=\\"http://somejsurl2.com\\"></script>
        <script src=\\"http://somejsurl3.com\\"></script>
        
    </body>
</html>"
`;

exports[`.document() - arguments given - should render template using values given 1`] = `
"<!doctype html>
<html lang=\\"en-NZ\\">
    <head>
        <meta charset=\\"utf-pretend-encoding\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link href=\\"http://somecssurl.com\\">
        
        <title>this goes in the title tag</title>
        this goes in the head section
    </head>
    <body>
        this goes in the body section
        <script src=\\"http://somejsurl.com\\"></script>
        
    </body>
</html>"
`;

exports[`.document() - js "type" is "esm" - should set type to module on script tags 1`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link href=\\"http://somecssurl1.com\\" type=\\"text/css\\">
        <link href=\\"http://somecssurl2.com\\" type=\\"text/css\\">
        <link href=\\"http://somecssurl3.com\\" type=\\"text/css\\">
        
        <title></title>
        
    </head>
    <body>
        
        <script src=\\"http://somejsurl1.com\\" type=\\"module\\"></script>
        <script src=\\"http://somejsurl2.com\\" type=\\"module\\"></script>
        <script src=\\"http://somejsurl3.com\\" type=\\"module\\"></script>
        
    </body>
</html>"
`;

exports[`.document() - js "type" is "esm" - should set type to module on script tags 2`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link href=\\"http://somecssurl1.com\\" type=\\"text/css\\">
        <script src=\\"http://somejsurl2.com/before\\" type=\\"module\\"></script>
        <title></title>
        
    </head>
    <body>
        
        <script src=\\"http://somejsurl3.com/after\\" type=\\"module\\"></script>
        <script type=\\"module\\">import(\\"http://somejsurl1.com/lazy\\")</script>
    </body>
</html>"
`;

exports[`.document() - no arguments given - should render template 1`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        
        
        <title></title>
        
    </head>
    <body>
        
        
        
    </body>
</html>"
`;
